src/parser/jobs/blu/modules/AlwaysBeCasting.tsx

```diff
@@ -0,0 +1,195 @@
+		}
+		const anyActionFilter = filter<Event>()
+			.source(this.parser.actor.id)
+			.action(noneOf([this.data.actions.PHANTOM_FLURRY.id]))
+			.type('action')
+		this.phantomFlurryInterruptingActionHook = this.addEventHook(anyActionFilter, this.onRemovePhantomFlurry)
+
+		this.phantomFlurryHistory.push(this.currentPhantomFlurry)
+	}
+
+	private onRemovePhantomFlurry(event: Events['statusRemove'] | Events['action']) {
+		if (this.currentPhantomFlurry == null) {
+			return
+		}
+		if (this.phantomFlurryInterruptingActionHook == null) {
+			return
+		}
+
+		this.currentPhantomFlurry.end = event.timestamp
+		this.removeEventHook(this.phantomFlurryInterruptingActionHook)
+
+		this.currentPhantomFlurry = undefined
+		this.phantomFlurryInterruptingActionHook = undefined
+
+	}
+
+	override considerCast(action: Action, castStart: number): boolean {
+		if (action === this.data.actions.PHANTOM_FLURRY) {
+			this.debug(`Phantom Flurry began channeling at ${this.parser.formatEpochTimestamp(castStart)}`)
+			return false
+		}
+
+		return super.considerCast(action, castStart)
+	}
+
+	override getUptimePercent(): number {
+		const fightDuration = this.parser.currentDuration - this.downtime.getDowntime()
+		const flurryDuration = this.phantomFlurryHistory.reduce((acc, flurry) => {
+			const downtime = this.downtime.getDowntime(
+				flurry.start,
+				flurry.end,
+			)
+			const phantomFlurryDurationOrGCD = Math.max(flurry.end - flurry.start, this.globalCooldown.getDuration())
+			return acc + phantomFlurryDurationOrGCD - downtime
+		}, 0)
+		const surpanakhaDuration = this.surpanakhas * SURPANAKHA_ANIMATION_LOCK_MS
+		const uptime = (this.gcdUptime + flurryDuration + surpanakhaDuration) / (fightDuration) * 100
+
+		return uptime
+	}
+
+	override onComplete() {
+		super.onComplete()
+
+		// Since we were already tracking Phantom Flurry, go ahead and take
+		// the chance to track if they dropped any damage ticks.
+		const missingFlurryTicks = this.phantomFlurryHistory
+			.reduce((acc, flurry) => {
+				const flurryChannelMs = flurry.end - flurry.start
+				const expectedFlurryChannel = (flurry.manualKick ? PHANTOM_FLURRY_CHANNEL_WITH_KICK_DURATION_MS : PHANTOM_FLURRY_CHANNEL_DURATION_MAX_MS)
+				const missingFlurryChannelMs = expectedFlurryChannel - flurryChannelMs
+				if (missingFlurryChannelMs <= 0) { return acc }
+
+				const missingTicks = Math.ceil(missingFlurryChannelMs / 1000)
+				return acc + missingTicks
+			}, 0)
+		this.suggestions.add(new TieredSuggestion({
+			icon: this.data.actions.PHANTOM_FLURRY.icon,
+			content: <Trans id="blu.phantom_flurry.dropped_ticks.content">
+				Dropping out of <DataLink action="PHANTOM_FLURRY" /> too early will lose damage ticks.  If you are in a <DataLink action="MOON_FLUTE" /> window you want to wait out the entire channel; if you are using it outside of a window and activating the final kick, wait until the last second the <DataLink status="PHANTOM_FLURRY" /> effect is active.
```

Comment:
Basically all of the rest of Xiva's copy does not use double-spaces after a period. Can we go to single-spaces for consistency?